code/_onclick/click_ctrl.dm

```diff
@@ -0,0 +1,123 @@
+/**
+ * Ctrl click
+ */
+/mob/proc/CtrlClickOn(atom/A)
+	base_click_ctrl(A)
+
+/**
+ * ### Base proc for ctrl click interaction left click.
+ *
+ * If you wish to add custom `click_ctrl` behavior for a single type, use that proc.
+ */
+/mob/proc/base_click_ctrl(atom/target)
+	SHOULD_NOT_OVERRIDE(TRUE)
+
+	// Check if they've hooked in to prevent src from ctrl clicking anything
+	if(SEND_SIGNAL(src, COMSIG_MOB_CTRL_CLICKED, target) & COMSIG_MOB_CANCEL_CLICKON)
+		return TRUE
+
+	var/can_use_click_action = FALSE
+	if(isturf(target))
+		// Turfs are special because they can't be used with can_perform_action
+		can_use_click_action = can_perform_turf_action(target)
+	else
+		can_use_click_action = can_perform_action(target, target.interaction_flags_click | SILENT_ADJACENCY)
+
+	. = TRUE
+	if(can_use_click_action)
+		// If it has a signal handler that returns a click action, done.
+		if(SEND_SIGNAL(target, COMSIG_CLICK_CTRL, src) & CLICK_ACTION_ANY)
+			return FALSE
```

Comment:
You should send this signal where you send COMSIG_MOB_CTRL_CLICKED to mimic past functionality.
This is likely going to break anything that relied on ctrl clicks to unconditionally work. Consider making a new signal to be sent after the `can_perform_action` checks have passed if you need it.

Reply:
I actually got the idea to send send the signal after `can_perform_action()` from `base_alt_click()`. 
https://github.com/tgstation/tgstation/blob/3817dd1fa9fddd7cb56dc8680b2103df4de8419b/code/_onclick/click_alt.dm#L36

So i should make a future PR to move the signal before `can_perform_action()` there as well